src/agent/scm/lib/svnwrapper.ts

```diff
@@ -0,0 +1,475 @@
+/// <reference path="../../definitions/node.d.ts"/>
+/// <reference path="../../definitions/toolrunner.d.ts"/>
+
+var tl = require('vso-task-lib');
+import events = require('events');
+import cm = require('../../common');
+import utilm = require('../../utilities');
+import Q = require('q');
+import os = require("os");
+var shell = require('shelljs');
+var path = require('path');
+var xmlrReader = require('xmlreader');
+
+import fs = require('fs');
+
+var administrativeDirectoryName = ".svn";
+
+export interface SvnMappingDetails {
+    serverPath: string;
+    localPath: string;
+    revision: string;
+    depth: string;
+    ignoreExternals: boolean;
+}
+
+export interface SvnWorkspace {
+    mappings: SvnMappingDetails[];
+}
+
+export interface ISvnMappingDictionary { [path: string]: SvnMappingDetails }
+
+export interface ISvnConnectionEndpoint {
+    username: string;
+    password: string;
+    realmName: string;
+    url: string;
+}
+
+export interface ISvnExecOptions {
+    cwd: string;
+    env: { [key: string]: string };
+    silent: boolean;
+    outStream: NodeJS.WritableStream;
+    errStream: NodeJS.WritableStream;
+    failOnStdErr: boolean;
+    ignoreReturnCode: boolean;
+}
+
+export class SvnWrapper extends events.EventEmitter {
+    constructor(ctx: cm.IExecutionContext) {
+        this.svnPath = shell.which('svn', false);
+        this.endpoint = <ISvnConnectionEndpoint>{};
+        this.ctx = ctx;
+        super();
+    }
+
+    public svnPath: string;
+    public endpoint: ISvnConnectionEndpoint;
+    public ctx: cm.IExecutionContext;
+
+    public setSvnConnectionEndpoint(endpoint: ISvnConnectionEndpoint) {
+        if (endpoint) {
+            this.endpoint = endpoint;
+        }
+    }
+
+    public getOldMappings(rootPath: string): Q.Promise<cm.IStringDictionary> {
+        if (shell.test("-f", rootPath)) {
+            throw new Error("The file " + rootPath + " already exists.");
+        }
+
+        if (shell.test("-d", rootPath)) {
+            
+            return this._getSvnWorkingCopyPaths(rootPath)
+            .then((workingDirectoryPaths: string[]) => {
+                if (workingDirectoryPaths) {
+        
+                    var mappingsPromise: Q.Promise<cm.IStringDictionary> = Q(<cm.IStringDictionary>{});
```

Comment:
Nitpick - it looks like we can declare this outside of the if block, eliminate the else at the end, and just return it.